pkg/scheduler/framework/v1alpha1/framework.go

```diff
@@ -627,6 +627,15 @@ func (f *framework) GetWaitingPod(uid types.UID) WaitingPod {
 	return f.waitingPods.get(uid)
 }
 
+// RemoveWaitingPod removes a WaitingPod given its UID.
+func (f *framework) RemoveWaitingPod(uid types.UID) {
+	waitingPod := f.waitingPods.get(uid)
+	if waitingPod != nil {
+		waitingPod.Reject("removed")
+		f.waitingPods.remove(uid)
```

Comment:
I think the rejected pod will be removed here.
https://github.com/kubernetes/kubernetes/blob/ef479c1a6fdedca9fa396dbcc19f89a983ac47a9/pkg/scheduler/framework/v1alpha1/framework.go#L588

Reply:
yes thank you @mrkm4ntr  ~ I know this. But I'm thinking about whether to delete it here to make sure the data is deleted.